j5/boards/sr/v4/power_board.py

```diff
@@ -0,0 +1,129 @@
+"""Classes for the SR v4 Power Board."""
+
+from enum import Enum
+from typing import TYPE_CHECKING, List, Union, cast
+
+from j5.backends import Backend, Environment
+from j5.boards import Board
+from j5.components import LED, BatterySensor, Button, Piezo, PowerOutput
+
+if TYPE_CHECKING:
+    from j5.components import (  # noqa: F401
+        Component,
+        ButtonInterface,
+        PowerOutputInterface,
+        PiezoInterface,
+        BatterySensorInterface,
+        LEDInterface,
+    )
+    from typing import Type  # noqa: F401
+
+
+class PowerOutputPosition(Enum):
+    """A mapping of name to number of the PowerBoard outputs."""
+
+    H0 = 0
+    H1 = 1
+    L0 = 2
+    L1 = 3
+    L2 = 4
+    L3 = 5
+
+
+PowerOutputGroupIndex = Union[int, PowerOutputPosition]
+
+
+class PowerOutputGroup:
+    """A group of PowerOutputs on the PowerBoard."""
+
+    def __init__(self, backend: Backend, board: Board):
+        self._backend = backend
+        self._board = board
+        self._outputs = [
```

Comment:
Should this be a Dictionary of Position to Output? The code relies on the fact the order of the elements in the list will stay the same. It's easy for a developer to sort a list and not realise they are changing behaviour but harder for them to change the keys in a dictionary and not realise they are doing something dodgy.

Not saying this has to change but those are my immediate thoughts

Reply:
I definitely would prefer this.

Reply:
Fixed in 176195f